modules/javafx.controls/src/main/java/javafx/scene/control/skin/TextAreaSkin.java

```diff
@@ -828,16 +820,29 @@ public Bounds getCaretBounds() {
             default: return super.queryAccessibleAttribute(attribute, parameters);
         }
     }
+    
+    @Override
+    public void install() {
+        super.install();
 
-    /** {@inheritDoc} */
-    @Override public void dispose() {
-        if (getSkinnable() == null) return;
-        getSkinnable().removeEventFilter(ScrollEvent.ANY, scrollEventFilter);
-        getChildren().remove(scrollPane);
-        super.dispose();
-
-        if (behavior != null) {
-            behavior.dispose();
+        // install default input map for the text area control
+        behavior = new TextAreaBehavior();
```

Comment:
I think, we can create the `behavior` object in the `Skin` constructor and invoke `Behavior.install()` from `Skin.install()` method.
It is possible to erroneously invoke this `install` method twice and we would end up with a leak.

Reply:
this makes sense, thanks!